### PR TITLE
jekyll: 3.0.1 -> 3.4.1

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by veprbl** [comment]
Ah, I guess this is broken:
```
$ nix-shell -p jekyll --command "jekyll new test"  
Running bundle install in /private/tmp/test... 
  Bundler: There was an error while trying to write to
  Bundler: `/nix/store/l67429rhvrmr7c4c1msb7s8zjq4fx7ad-gemfile-and-lockfile/.bundle/config`.
  Bundler: It is likely that you need to grant write permissions for that path.
```

File: pkgs/development/ruby-modules/gem-config/default.nix [hunk 1]
```diff
@@ -78,6 +78,13 @@ in
         [ darwin.apple_sdk.frameworks.CoreServices ];
   };
 
+  jekyll = attrs: {
+    postInstall = ''
+      installPath=$(cat $out/nix-support/gem-meta/install-path)
+      sed -i $installPath/lib/jekyll/commands/new.rb -e 's@Exec.run("bundle", "install"@Exec.run("nix-shell", "-p", "bundler", "--pure", "--run", "${bundler}/bin/bundle install"@'
```
Comment (league) [inline pkgs/development/ruby-modules/gem-config/default.nix]: Ah, that's clever. I suppose there's no problem with nested nix-shell environments? (I never tried, AFAIK.)

Comment (veprbl) [inline pkgs/development/ruby-modules/gem-config/default.nix]: @league I think this exec should be disabled altogether. Nothing that user can specify in that gemfile will propagate to nix'ed jekyll. User would have to add stuff to override pkgs/applications/misc/jekyll/Gemfile . What do you think?

Comment (veprbl) [inline pkgs/development/ruby-modules/gem-config/default.nix]: For example user in #20617 needs an additional Gem and there is no straightforward way to add it.

**Comment by veprbl** [comment]
Basic functionality works in my tests now. There is still a problem with building sites that depend on additional plugins (e.g. #20617), but it is there already and this PR has nothing to do with it.

**Comment by league** [comment]
LGTM – I don't immediately see a better way to handle the bundler issue. The fact that `jekyll new` now generates its own `Gemfile` makes things a little complicated for NixOS – but at least with this solution the basic tutorials for building a new site with default plugins will work. Maybe one day, commonly-used plugins could be optional arguments to the nix expression…?

**Comment by league** [comment]
@veprbl Now that I see how one patches up the installed Ruby code, this small addition to gem-config would also address #22858… but either way, I'll see how far I get upstream.

```nix
  # disable bundle install as it can't install anything in addition to what is
  # specified in pkgs/applications/misc/jekyll/Gemfile anyway. Also do chmod_R
  # to compensate for read-only files in site_template in nix store.
  jekyll = attrs: {
    postInstall = ''
      installPath=$(cat $out/nix-support/gem-meta/install-path)
      sed -i $installPath/lib/jekyll/commands/new.rb \
          -e 's@Exec.run("bundle", "install"@Exec.run("true"@' \
          -e 's@FileUtils.cp_r site_template + "/.", path@FileUtils.cp_r site_template + "/.", path; FileUtils.chmod_R "u+w", path@'
    '';
  };
```

**Comment by veprbl** [comment]
@league ```cp_r``` has ```:preserve``` argument. Can't find docs for it though.

**Comment by veprbl** [comment]
Seems to not be able to do what we want. The problem seems to be standard: how to get files at 644 and directories at 755. I don't know if this can be done beautifully enough to upstream.

**Comment by league** [comment]
@veprbl Right. I also read (not under `cp_r` doc, but for `copy_entry`) that "If preserve is true, this method preserves owner, group, and modified time. Permissions are copied regardless preserve." https://ruby-doc.org/stdlib-2.2.2/libdoc/fileutils/rdoc/FileUtils.html

**Comment by veprbl** [comment]
This call for a ruby wizard :)

**Comment by veprbl** [comment]
this is good to be merged btw

**Pull request merged**